src/Components/test/E2ETest/Tests/ClientSideHostingTest.cs

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure;
+using Microsoft.AspNetCore.Components.E2ETest.Infrastructure.ServerFixtures;
+using Microsoft.AspNetCore.E2ETesting;
+using Microsoft.AspNetCore.Hosting;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Support.UI;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    /// <summary>
+    /// Tests for various MapFallbackToClientSideBlazor overloads. We're just verifying that things render correctly.
+    /// That means that routing and file serving is working for the startup pattern under test.
+    /// </summary>
+    public class ClientSideHostingTest : ServerTestBase<AspNetSiteServerFixture>
+    {
+        public ClientSideHostingTest(
+            BrowserFixture browserFixture,
+            AspNetSiteServerFixture serverFixture,
+            ITestOutputHelper output)
+            : base(browserFixture, serverFixture, output)
+        {
+            serverFixture.ApplicationAssembly = typeof(TestServer.Program).Assembly;
+            serverFixture.BuildWebHostMethod = BuildWebHost;
+        }
+
+        private IWebHost BuildWebHost(string[] args)
+        {
+            return TestServer.Program.BuildWebHost<TestServer.StartupWithMapFallbackToClientSideBlazor>(args);
+        }
+
+        [Fact]
+        public void MapFallbackToClientSideBlazor_FilePath()
+        {
+            Navigate("/filepath", noReload: true);
+            WaitUntilLoaded();
+            Assert.NotNull(Browser.FindElement(By.Id("test-selector")).Text);
```

Comment:
This assertion is a bit puzzling to me. Why check the nullness of the element's text property? Why are we checking something about the test selector anyway?

I would have expected something like checking that the "Hello, world!" heading has been rendered instead.

Same with other tests below.

Reply:
Sure, that's fine. I just wanted to make it something that won't change.

Reply:
This page doesn't have a heading, it just has that select. I'm updating this to just drop `.Text` to make it clear that it's about seeing that element come up. I chose this because it's used in so many tests - which makes it very unlikely these tests would break without something going wrong.